supervisor/archives_api.go

```diff
@@ -92,17 +92,25 @@ func (self ArchiveAPI) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			return
 		}
 
+		taskChan := make(chan string)
 		// tell the supervisor to schedule a task
 		self.Tasks <- &db.Task{
-			Op:          db.RestoreOperation,
-			Owner:       params.Owner,
-			TargetUUID:  tid,
-			ArchiveUUID: id,
-			RestoreKey:  archive.StoreKey,
+			Op:           db.RestoreOperation,
+			Owner:        params.Owner,
+			TargetUUID:   tid,
+			ArchiveUUID:  id,
+			RestoreKey:   archive.StoreKey,
+			TaskUUIDChan: taskChan,
 		}
 
-		w.WriteHeader(200)
-		JSONLiteral(w, fmt.Sprintf(`{"ok":"scheduled"}`))
+		var taskUUID string = <-taskChan
```

Comment:
This should be a `:=` assignment without the `var` keyword or `string` type declaration